### ISSUE TITLE
Add admin powers to access locked data from the api

### ISSUE THREAD

**Issue by Shadowys** [Shadowys/hako#31]
As of now, the api backend is completely open. However, for locked content (as in locked by an admin or owner), the api should require the caller to specify their id and password to access the data.

**Comment by viggy96** [comment]
How do we plan on storing passwords? This is really my one pet peeve. I'd recommend the use of some sort of API like Firebase or something. With the use of these, we'd just be storing tokens which stand-in for both the user and password, protecting their privacy. 

I mean, if Google or Facebook's security is compromised, the world has bigger problems.

**Comment by Shadowys** [comment]
Would this mean that we move user authentication outside of our database? I would prefer this as it will decouple our api with authentication.

**Comment by viggy96** [comment]
Yeah, pretty much. We'd still need an account database, keeping track of the rights of various users, but the only identifying information would be the user token.

**Comment by Shadowys** [comment]
Do you have example code on how should we be doing that?

**Comment by viggy96** [comment]
There's nothing special we need for the backend. Just a user table with a column for user tokens. We would obviously need more columns for purposes of keeping track of which user has rights where, but that has nothing to do with the authentication API.
